src/plugins/data/public/query/state_sync/connect_to_query_state.ts

```diff
@@ -91,15 +91,21 @@ export const connectToQueryState = <S extends QueryState>(
     } else if (syncConfig.filters === FilterStateStore.GLOBAL_STATE) {
       if (
         !initialState.filters ||
-        !compareFilters(initialState.filters, filterManager.getGlobalFilters(), COMPARE_ALL_OPTIONS)
+        !compareFilters(initialState.filters, filterManager.getGlobalFilters(), {
```

Comment:
Seems we have the state information inside `FilterManager`, as filters are being correctly returned from `getGlobalFilters` and `getAppFilters`. Why not update the $state value before returning the filters?